From 142733a20383f633831d7b9d7f89a1807ab1ad66 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Thu, 2 Oct 2014 08:28:30 -0400
Subject: [tools] perf: Shorten sample symbol resolving function signature

Message-id: <20141002082830.GE9764@krava.brq.redhat.com>
Patchwork-id: 96875
O-Subject: [PATCHv2 RHEL7.1 BZ1134356 043/307] perf tools: Shorten sample symbol resolving function signature
Bugzilla: 1134356
RH-Acked-by: Don Zickus <dzickus@redhat.com>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit e80faac0460f178a5be576b4260897f997109e73
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date:   Wed Jan 22 13:05:06 2014 -0300

description
===========
Since three of the parameters come from the same 'struct addr_location',
rename machine__resolve_mem() to sample__resolve_mem() and pass the
that addr_location instead.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-report.c b/tools/perf/builtin-report.c
index 08f0187..f0888ba 100644
--- a/tools/perf/builtin-report.c
+++ b/tools/perf/builtin-report.c
@@ -88,7 +88,7 @@ static int report__add_mem_hist_entry(struct perf_tool *tool, struct addr_locati
  if (err)
   return err;
 
- mi = machine__resolve_mem(al->machine, al->thread, sample, al->cpumode);
+ mi = sample__resolve_mem(sample, al);
  if (!mi)
   return -ENOMEM;
 
diff --git a/tools/perf/util/hist.c b/tools/perf/util/hist.c
index e4e6249..6240ca4 100644
--- a/tools/perf/util/hist.c
+++ b/tools/perf/util/hist.c
@@ -369,7 +369,7 @@ static struct hist_entry *add_hist_entry(struct hists *hists,
    he_stat__add_period(&he->stat, period, weight);
 
    /*
-    * This mem info was allocated from machine__resolve_mem
+    * This mem info was allocated from sample__resolve_mem
     * and will not be used anymore.
     */
    zfree(&entry->mem_info);
diff --git a/tools/perf/util/machine.c b/tools/perf/util/machine.c
index 620a198..45fd72f 100644
--- a/tools/perf/util/machine.c
+++ b/tools/perf/util/machine.c
@@ -1238,18 +1238,17 @@ static void ip__resolve_data(struct machine *machine, struct thread *thread,
  ams->map = al.map;
 }
 
-struct mem_info *machine__resolve_mem(struct machine *machine,
-          struct thread *thr,
-          struct perf_sample *sample,
-          u8 cpumode)
+struct mem_info *sample__resolve_mem(struct perf_sample *sample,
+         struct addr_location *al)
 {
  struct mem_info *mi = zalloc(sizeof(*mi));
 
  if (!mi)
   return NULL;
 
- ip__resolve_ams(machine, thr, &mi->iaddr, sample->ip);
- ip__resolve_data(machine, thr, cpumode, &mi->daddr, sample->addr);
+ ip__resolve_ams(al->machine, al->thread, &mi->iaddr, sample->ip);
+ ip__resolve_data(al->machine, al->thread, al->cpumode,
+    &mi->daddr, sample->addr);
  mi->data_src.val = sample->data_src;
 
  return mi;
diff --git a/tools/perf/util/machine.h b/tools/perf/util/machine.h
index f77e91e..9ddacd9 100644
--- a/tools/perf/util/machine.h
+++ b/tools/perf/util/machine.h
@@ -94,9 +94,8 @@ void machine__delete(struct machine *machine);
 struct branch_info *machine__resolve_bstack(struct machine *machine,
          struct thread *thread,
          struct branch_stack *bs);
-struct mem_info *machine__resolve_mem(struct machine *machine,
-          struct thread *thread,
-          struct perf_sample *sample, u8 cpumode);
+struct mem_info *sample__resolve_mem(struct perf_sample *sample,
+         struct addr_location *al);
 int machine__resolve_callchain(struct machine *machine,
           struct perf_evsel *evsel,
           struct thread *thread,
-- 
1.7.1